### PR TITLE
V3.0.0

### PR DESCRIPTION
**Pull request by mbarrenechea**
This PR starts as a change for adding support video tiles, but we realize that we needed to change the structure of layers for better performance and better understanding. Now we have a clear differentiation between source and render, in other words, where to fetch/get the data and how to display it.

If you want to use this version run this command
```
yarn add layer-manager@beta-3.0
```
It's very important to use these versions
- `deck.gl@7.3.6`
- `luma.gl@7.3.2`
- `viewport-mercator-project@6.1.1`

Do we really care about Leafet? Who is going to migrate it?

TODO:
- [x] Check vector layers
- [x] Check raster layers
- [x] Check geojson layers
- [x] Check decoded layers
- [x] Documentation
- [ ] Tests
- [ ] Leaflet migration?

### CODE REVIEW

File: src/components/layer.js [hunk 1]
```diff
@@ -59,13 +78,34 @@ class Layer extends PureComponent {
...
+      prevRender.parse === false
+        ? prevRender
+        : JSON.parse(replace(JSON.stringify(prevRender), prevParams, prevSqlParams));
+
+    const renderParsed =
+      render.parse === false
+        ? render
+        : JSON.parse(replace(JSON.stringify(render), params, sqlParams));
+
+    if (sourceParsed && !isEqual(sourceParsed, prevSourceParsed)) {
```
Comment (sorodrigo) [inline src/components/layer.js]: Ideally we should save the source after parsing, this way we can do a simple : `Object.is(source, prevSource)` or a shallow compare:
```js
Object.keys(source).forEach(key => Object.is(source[key], prevSource[key])
```

File: src/plugins/plugin-mapbox-gl/custom-layers/bitmap-layer/bitmap-layer-fragment.js [hunk 2]
```diff
@@ -9,32 +9,33 @@ uniform sampler2D bitmapTexture;
```
Comment (edbrett) [inline src/plugins/plugin-mapbox-gl/custom-layers/bitmap-layer/bitmap-layer-fragment.js]: Let remove this layer if it is no longer being used.

File: src/components/layer.js (hunk 1 above)
Comment (mbarrenechea) [reply src/components/layer.js]: As we talk yesterday, we can't do this at this point. We should save it in the model parsed, that's true. We will think of a way of doing it

File: src/plugins/plugin-mapbox-gl/custom-layers/bitmap-layer/bitmap-layer-fragment.js (hunk 2 above)
Comment (mbarrenechea) [reply src/plugins/plugin-mapbox-gl/custom-layers/bitmap-layer/bitmap-layer-fragment.js]: It was created for the first approach of video tiles, but as we are going to use the second approach (even if that means that we are going to fork a version of mapbox) I will remove all the dependencies

Comment (davidsingal) [reply src/plugins/plugin-mapbox-gl/custom-layers/bitmap-layer/bitmap-layer-fragment.js]: Try to avoid forks, it means more to maintenance

Comment (mbarrenechea) [reply src/plugins/plugin-mapbox-gl/custom-layers/bitmap-layer/bitmap-layer-fragment.js]: We can help the guy creating a PR to Mapbox, meanwhile, there is no other option...

**Comment by sorodrigo** [comment]
I think Leaflet support is nice to have but doesn't have to be supported right now, can be added later.

**Pull request merged**